bazel build failed on ppc64le

Please provide the following information. The more we know about your system and use case, the more easily and likely we can help.

### Description of the problem / feature request / question:

_If asking a question or requesting a feature, also tell us about the underlying problem you're trying to solve._


### If possible, provide a minimal example to reproduce the problem:


### Environment info

* Operating System:
```
root@cit1046:~/go/src/istio.io/pilot# uname -a
Linux cit1046.rtp.raleigh.ibm.com 4.4.0-79-generic #100-Ubuntu SMP Wed May 17 19:56:58 UTC 2017 ppc64le ppc64le ppc64le GNU/Linux
```

BTW: I build `bazel 0.5.4` myself for ppc64le.

* Bazel version (output of `bazel info release`):
```
root@cit1046:~/go/src/istio.io/pilot# bazel version
Build label: 0.5.4- (@non-git)
Build target: bazel-out/local-opt/bin/src/main/java/com/google/devtools/build/lib/bazel/BazelServer_deploy.jar
Build time: Tue Sep 26 04:10:02 2017 (1506399002)
Build timestamp: 1506399002
Build timestamp as int: 1506399002
```


* If `bazel info release` returns "development version" or "(@non-git)", please tell us what source tree you compiled Bazel from; git commit hash is appreciated (`git rev-parse HEAD`):


###  Have you found anything relevant by searching the web?
(e.g. [StackOverflow answers](http://stackoverflow.com/questions/tagged/bazel),
[GitHub issues](https://github.com/bazelbuild/bazel/issues),
email threads on the [`bazel-discuss`](https://groups.google.com/forum/#!forum/bazel-discuss) Google group)


### Anything else, information or logs or outputs that would be helpful?
(If they are large, please upload as attachment or provide link).

1) Get the istio pilot code.
```
git clone git@github.com:istio/pilot.git
```

2) Build the source code with `make setup`

```
$ make setup
....
WARNING: /root/.cache/bazel/_bazel_root/1745129434edd1c4aecf285cdde2d234/external/bazel_tools/tools/build_defs/pkg/pkg.bzl:196:9: pkg_tar: renaming non-dict `files` attribute to `srcs`.
ERROR: /root/go/src/istio.io/pilot/test/client/BUILD:3:1: no such package '@com_github_gorilla_websocket//': no such package '@io_bazel_rules_go_repository_tools//': failed to build gazelle: /root/.cache/bazel/_bazel_root/1745129434edd1c4aecf285cdde2d234/external/go1_8_3_linux_amd64/bin/go: 1: /root/.cache/bazel/_bazel_root/1745129434edd1c4aecf285cdde2d234/external/go1_8_3_linux_amd64/bin/go: uE@p@8: not found
/root/.cache/bazel/_bazel_root/1745129434edd1c4aecf285cdde2d234/external/go1_8_3_linux_amd64/bin/go: 2: /root/.cache/bazel/_bazel_root/1745129434edd1c4aecf285cdde2d234/external/go1_8_3_linux_amd64/bin/go: @$: not found
/root/.cache/bazel/_bazel_root/1745129434edd1c4aecf285cdde2d234/external/go1_8_3_linux_amd64/bin/go: 1: /root/.cache/bazel/_bazel_root/1745129434edd1c4aecf285cdde2d234/external/go1_8_3_linux_amd64/bin/go: ELF: not found
/root/.cache/bazel/_bazel_root/1745129434edd1c4aecf285cdde2d234/external/go1_8_3_linux_amd64/bin/go: 1: /root/.cache/bazel/_bazel_root/1745129434edd1c4aecf285cdde2d234/external/go1_8_3_linux_amd64/bin/go: �: not found
/root/.cache/bazel/_bazel_root/1745129434edd1c4aecf285cdde2d234/external/go1_8_3_linux_amd64/bin/go: 1: /root/.cache/bazel/_bazel_root/1745129434edd1c4aecf285cdde2d234/external/go1_8_3_linux_amd64/bin/go: �: not found
/root/.cache/bazel/_bazel_root/1745129434edd1c4aecf285cdde2d234/external/go1_8_3_linux_amd64/bin/go: 1: /root/.cache/bazel/_bazel_root/1745129434edd1c4aecf285cdde2d234/external/go1_8_3_linux_amd64/bin/go: �{�{k�: not found
/root/.cache/bazel/_bazel_root/1745129434edd1c4aecf285cdde2d234/external/go1_8_3_linux_amd64/bin/go: 1: /root/.cache/bazel/_bazel_root/1745129434edd1c4aecf285cdde2d234/external/go1_8_3_linux_amd64/bin/go: �: not found
/root/.cache/bazel/_bazel_root/1745129434edd1c4aecf285cdde2d234/external/go1_8_3_linux_amd64/bin/go: 1: /root/.cache/bazel/_bazel_root/1745129434edd1c4aecf285cdde2d234/external/go1_8_3_linux_amd64/bin/go: L: not found
/root/.cache/bazel/_bazel_root/1745129434edd1c4aecf285cdde2d234/external/go1_8_3_linux_amd64/bin/go: 1: /root/.cache/bazel/_bazel_root/1745129434edd1c4aecf285cdde2d234/external/go1_8_3_linux_amd64/bin/go: �: not found
/root/.cache/bazel/_bazel_root/1745129434edd1c4aecf285cdde2d234/external/go1_8_3_linux_amd64/bin/go: 1: /root/.cache/bazel/_bazel_root/1745129434edd1c4aecf285cdde2d234/external/go1_8_3_linux_amd64/bin/go: G�{�: not found
/root/.cache/bazel/_bazel_root/1745129434edd1c4aecf285cdde2d234/external/go1_8_3_linux_amd64/bin/go: 1: /root/.cache/bazel/_bazel_root/1745129434edd1c4aecf285cdde2d234/external/go1_8_3_linux_amd64/bin/go: cannot open 0B�0BN
                                                   BHB�HBN�
                                                          Q���o@E�@ENH
                                                                      ^���o�E��ENP: No such file
/root/.cache/bazel/_bazel_root/1745129434edd1c4aecf285cdde2d234/external/go1_8_3_linux_amd64/bin/go: 1: /root/.cache/bazel/_bazel_root/1745129434edd1c4aecf285cdde2d234/external/go1_8_3_linux_amd64/bin/go: -R: not found
/root/.cache/bazel/_bazel_root/1745129434edd1c4aecf285cdde2d234/external/go1_8_3_linux_amd64/bin/go: 2: /root/.cache/bazel/_bazel_root/1745129434edd1c4aecf285cdde2d234/external/go1_8_3_linux_amd64/bin/go:F�FN�
                                     ��FNm4@H�@HN: not found
/root/.cache/bazel/_bazel_root/1745129434edd1c4aecf285cdde2d234/external/go1_8_3_linux_amd64/bin/go: 1: /root/.cache/bazel/_bazel_root/1745129434edd1c4aecf285cdde2d234/external/go1_8_3_linux_amd64/bin/go: JN: not found
/root/.cache/bazel/_bazel_root/1745129434edd1c4aecf285cdde2d234/external/go1_8_3_linux_amd64/bin/go: 2: /root/.cache/bazel/_bazel_root/1745129434edd1c4aecf285cdde2d234/external/go1_8_3_linux_amd64/bin/go: ���: not found
/root/.cache/bazel/_bazel_root/1745129434edd1c4aecf285cdde2d234/external/go1_8_3_linux_amd64/bin/go: 4: /root/.cache/bazel/_bazel_root/1745129434edd1c4aecf285cdde2d234/external/go1_8_3_linux_amd64/bin/go:O�M��MN�: not found
/root/.cache/bazel/_bazel_root/1745129434edd1c4aecf285cdde2d234/external/go1_8_3_linux_amd64/bin/go: 4: /root/.cache/bazel/_bazel_root/1745129434edd1c4aecf285cdde2d234/exc��Nm���Nk: not foundamd64/bin/go: Y�����Np
/root/.cache/bazel/_bazel_root/1745129434edd1c4aecf285cdde2d234/external/go1_8_3_linux_amd64/bin/go: 1: /root/.cache/bazel/_bazel_root/1745129434edd1c4aecf285cdde2d234/external/go1_8_3_linux_amd64/bin/go: Syntax error: Unterminated quoted string
 and referenced by '//test/client:go_default_library'.
ERROR: Analysis of target '//test/client:client' failed; build aborted.
INFO: Elapsed time: 55.074s
Makefile:19: recipe for target 'setup' failed
make: *** [setup] Error 1
```

@jayconrod If i try this inside the istio project I get the following:

```
istio# BAZEL_BUILD_ARGS=--experimental_platforms=@io_bazel_rules_go//go/toolchain:linux_ppc64le make build                     
touch pilot/platform/kube/config    
bazel  build --experimental_platforms=@io_bazel_rules_go//go/toolchain:linux_ppc64le //...                                                        
ERROR: While resolving toolchains for target //mixer/tools/codegen/pkg/modelgen:go_default_library: no matching toolchains found for types @io_bazel_rules_go//go:toolchain.
ERROR: Analysis of target '//mixer/tools/codegen/pkg/modelgen:go_default_library' failed; build aborted: no matching toolchains found for types @io_bazel_rules_go//go:toolchain.
INFO: Elapsed time: 27.829s
Makefile:124: recipe for target 'build' failed
make: *** [build] Error 1
```

I haven't dug into it yet but I thought I'd post in case it's obvious to you what's going on there.

Thans @jayconrod and @clnperez 

Just did a test with @jayconrod 's patch and steps here https://github.com/bazelbuild/rules_go/pull/1145 , failed as following:

```
root@cit1046:~/go/src/istio.io/istio# BAZEL_BUILD_ARGS=--experimental_platforms=@io_bazel_rules_go//go/toolchain:linux_ppc64le make build
touch pilot/platform/kube/config
bazel  build --experimental_platforms=@io_bazel_rules_go//go/toolchain:linux_ppc64le //...
...............................
DEBUG: /root/.cache/bazel/_bazel_root/3a0d045513e496afb4d4a4ec4724ce17/external/io_bazel_rules_go/proto/def.bzl:100:3: You no longer need to call proto_register_toolchains(), it does nothing
DEBUG: /root/.cache/bazel/_bazel_root/3a0d045513e496afb4d4a4ec4724ce17/external/io_bazel_rules_go/go/private/sdk.bzl:99:5: bin/go: 1: bin/go: ELF: not found
bin/go: 2: bin/go: cannot open : No such file
bin/go: 2: bin/go: @$: not found
bin/go: 2: bin/go: �-�-�i����@*�v: not found
bin/go: 2: bin/go: : not found
: not foundbin/go: Z
bin/go: 1: bin/go: ��: not foundbin/go: 2: bin/go: cannot open G: No such file

: not foundbin/go:
bin/go: 2: bin/go: cannot open ]�: No such file
bin/go: 2: bin/go: : not found
bin/go: 3: bin/go:@��@�P�

                        �P{B�����P:
                                   �����Px: not found
bin/go: 4: bin/go: Syntax error: "(" unexpected
ERROR: /root/.cache/bazel/_bazel_root/3a0d045513e496afb4d4a4ec4724ce17/external/org_pubref_rules_protobuf/gogo/BUILD:29:1: no such package '@com_github_gogo_protobuf//protoc-gen-gogoslick': no such package '@io_bazel_rules_go_repository_tools//': no such package '@go_sdk//': Traceback (most recent call last):
	File "/root/.cache/bazel/_bazel_root/3a0d045513e496afb4d4a4ec4724ce17/external/io_bazel_rules_go/go/private/sdk.bzl", line 47
		_prepare(ctx)
	File "/root/.cache/bazel/_bazel_root/3a0d045513e496afb4d4a4ec4724ce17/external/io_bazel_rules_go/go/private/sdk.bzl", line 100, in _prepare
		fail("failed to list standard package...")
failed to list standard packages and referenced by '@org_pubref_rules_protobuf//gogo:gogoslick'
ERROR: Analysis of target '//mixer/tools/codegen/pkg/interfacegen/testdata:apa' failed; build aborted: Loading failed
INFO: Elapsed time: 81.652s
FAILED: Build did NOT complete successfully (419 packages loaded)
Makefile:124: recipe for target 'build' failed
make: *** [build] Error 1
root@cit1046:~/go/src/istio.io/istio# go
```

```
root@cit1046:~/go/src/istio.io/istio# make build
bazel  build  //...
DEBUG: /root/.cache/bazel/_bazel_root/3a0d045513e496afb4d4a4ec4724ce17/external/io_bazel_rules_go/proto/def.bzl:100:3: You no longer need to call proto_register_toolchains(), it does nothing
DEBUG: /root/.cache/bazel/_bazel_root/3a0d045513e496afb4d4a4ec4724ce17/external/io_bazel_rules_go/go/private/sdk.bzl:99:5: bin/go: 1: bin/go: ELF: not found
bin/go: 2: bin/go: cannot open : No such file
bin/go: 2: bin/go: @$: not found
bin/go: 2: bin/go: : not found
bin/go: 2: bin/go: �-�-�i����@*�v: not found
: not foundbin/go: Z
bin/go: 2: bin/go: cannot open G: No such file
: not foundbin/go:
bin/go: 1: bin/go: ��: not found
bin/go: 2: bin/go: cannot open ]�: No such file
bin/go: 2: bin/go: : not found
bin/go: 3: bin/go:@��@�P�

                        �P{B�����P:
                                   �����Px: not found
bin/go: 4: bin/go: Syntax error: "(" unexpected
ERROR: /root/.cache/bazel/_bazel_root/3a0d045513e496afb4d4a4ec4724ce17/external/org_pubref_rules_protobuf/gogo/BUILD:29:1: no such package '@com_github_gogo_protobuf//protoc-gen-gogoslick': no such package '@io_bazel_rules_go_repository_tools//': no such package '@go_sdk//': Traceback (most recent call last):
	File "/root/.cache/bazel/_bazel_root/3a0d045513e496afb4d4a4ec4724ce17/external/io_bazel_rules_go/go/private/sdk.bzl", line 47
		_prepare(ctx)
	File "/root/.cache/bazel/_bazel_root/3a0d045513e496afb4d4a4ec4724ce17/external/io_bazel_rules_go/go/private/sdk.bzl", line 100, in _prepare
		fail("failed to list standard package...")
failed to list standard packages and referenced by '@org_pubref_rules_protobuf//gogo:gogoslick'
ERROR: Analysis of target '//mixer/template/reportnothing:go_default_library_gogo_proto' failed; build aborted: Loading failed
INFO: Elapsed time: 72.177s
FAILED: Build did NOT complete successfully (3 packages loaded)
Makefile:124: recipe for target 'build' failed
make: *** [build] Error 1
```

--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -9,7 +9,7 @@ module(
 # conflicting with the real bazel_features repo.
 bazel_dep(name = "bazel_features", version = "1.9.1", repo_name = "io_bazel_rules_go_bazel_features")
 bazel_dep(name = "bazel_skylib", version = "1.2.0")
-bazel_dep(name = "platforms", version = "0.0.4")
+bazel_dep(name = "platforms", version = "0.0.10")
 bazel_dep(name = "rules_proto", version = "6.0.0")
 bazel_dep(name = "protobuf", version = "3.19.2", repo_name = "com_google_protobuf")
 

--- a/go/private/platforms.bzl
+++ b/go/private/platforms.bzl
@@ -31,7 +31,7 @@ BAZEL_GOARCH_CONSTRAINTS = {
     "arm": "@platforms//cpu:armv7",
     "arm64": "@platforms//cpu:aarch64",
     "ppc64": "@platforms//cpu:ppc",
-    "ppc64le": "@platforms//cpu:ppc",
+    "ppc64le": "@platforms//cpu:ppc64le",
     "s390x": "@platforms//cpu:s390x",
 }
 

--- a/go/private/repositories.bzl
+++ b/go/private/repositories.bzl
@@ -258,6 +258,16 @@ def go_rules_dependencies(force = False):
         patch_args = ["-p1"],
     )
 
+    wrapper(
+        http_archive,
+        name = "platforms",
+        sha256 = "218efe8ee736d26a3572663b374a253c012b716d8af0c07e842e82f238a0a7ee",
+        urls = [
+            "https://mirror.bazel.build/github.com/bazelbuild/platforms/releases/download/0.0.10/platforms-0.0.10.tar.gz",
+            "https://github.com/bazelbuild/platforms/releases/download/0.0.10/platforms-0.0.10.tar.gz",
+        ],
+    )
+
     # releaser:upgrade-dep golang mock
     _maybe(
         http_archive,

